files/SZWEEXEC/ZWEGEN00

```diff
@@ -0,0 +1,890 @@
+    Use ConfigMgr to validate the current configuration.
+================================================================================
+*/
+
+if Validate(schemaChain, configChain) > 0 then do
+  ExitWithRC(8)
+end
+
+/*
+================================================================================
+    If we should generate jcl, then use ConfigMgr to get the configuration
+    values.
+================================================================================
+*/
+
+if COMPARE(operation, 'nogenerate') = 0 then do
+  exit 0
+end
+else do
+  if GetConfiguration() > 0 then do
+    ExitWithRC(8)
+  end
+end
+
+/*
+================================================================================
+    Prepare header information ahead of time so that it gets substituted in
+    each member.
+================================================================================
+*/
+
+CFG.zwe.header.user = USERID()
+CFG.zwe.header.date = TRANSLATE(DATE(), '-', ' ')
+CFG.zwe.header.time = TIME()
+CFG.ZWE_CLI_PARAMETER_CONFIG = configChainWithMembers
+
+/*
+================================================================================
+    Create a data set with attributes like the original jcl library and copy
+    all the members of esm jcl.
+================================================================================
+*/
+
+jcl = CFG.zowe.setup.dataset.prefix'.SZWESAMP'
+jclCopy = CFG.zowe.setup.dataset.jcllib
+
+say 'Creating a fresh copy of 'jcl' named 'jclCopy'.'
+
+x = DeleteDataSet(jclCopy)
+
+if CreatePartitionedDataSet(jclCopy, 80, 'f,b') > 0 then do
+  ExitWithRC(8)
+end
+
+if AllocateDataSet(jclCopy, 'zweto') > 0 then do
+  ExitWithRC(8)
+end
+
+if AllocateDataSet(jcl, 'zwefrom') > 0 then do
+  ExitWithRC(8)
+end
+
+if GetDataIDFromDD('zwefrom') > 0 then do
+  ExitWithRC(8)
+end
+
+zwefrid = !dataid
+
+if GetDataIDFromDD('zweto') > 0 then do
+  ExitWithRC(8)
+end
+
+zwetoid = !dataid
+
+if CopyDataSetByDataID(zwefrid, zwetoid) > 0 then do
+  ExitWithRC(8)
+end
+
+if FreeDataID(zwefrid) > 0 then do
+  ExitWithRC(8)
+end
+
+if FreeDataID(zwetoid) > 0 then do
+  ExitWithRC(8)
+end
+
+if FreeByDD('zwefrom') > 0 then do
+  ExitWithRC(8)
+end
+
+if FreeByDD('zweto') > 0 then do
+  ExitWithRC(8)
+end
+
+
+/* members which are not JCL */
+x = DeleteDataSet(jclCopy'(ZWEGENER)')
+x = DeleteDataSet(jclCopy'(ZWESIP00)')
+x = DeleteDataSet(jclCopy'(ZWESIPRG)')
+x = DeleteDataSet(jclCopy'(ZWESISCH)')
+x = DeleteDataSet(jclCopy'(ZWESECKG)')
+x = DeleteDataSet(jclCopy'(ZWEINSTL)')
+
+
+/*
+================================================================================
+    Determine the external security manager on the system so that the correct
+    jcl templates can be used.
+================================================================================
+*/
+
+CVT_ADDR = C2X(STORAGE(D2X(16), 4))
+CVTRAC_ADDR = C2X(STORAGE(D2X(X2D(CVT_ADDR) + 992), 4))
+CVTRAC_VAL = STORAGE(CVTRAC_ADDR, 4)
```

Comment:
I was trying to generate all security jobs on the same LPAR (TSS), however it did not work.
This detection ignores the `zowe.setup.security.product` resulting in:
- `ZWENOSEC` is using value `zowe.setup.security.product` from `zowe.yaml`
- But the security/certificate members depending on `esm` were always TSS (because of lines 179-220).
- Probably fine on production level
- But during the testing and trying on dev/mini system, you would like to review JCL for other `esm` without need to run it on the system with specific `esm`

Reply:
Do you know how to use the value of `zowe.setup.security.product` if it exists, but if not default to this?
Because, every user would only have one ESM right? So, asking them which one seems unnecessary if we can detect it. Overriding the detection only makes sense for development I think.

Reply:
For my testing purposes, I have modified to `ZWEGEN00` to accept third parameter `esm`. If `esm` was set, the detection was skipped. But I don't see this as a problem on production. Anyway, if this will be required in the future by customers, the "fix" is just couple lines.